test/images/vulnerability-scan/scan-image.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Pipeline Service Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+
+VULNERABILITIES_EXIST=1
+
+get_digest() {
+  #number of retries=3
+  for _ in {1..3}
+  do
+    #Get the digest of the image
+    digest_http_code=$(curl -s -w '%{http_code}' -o /tmp/digest.json -H "Accept: application/vnd.quay.distribution.manifest.v2+json Content-type: application/json Authorization: Bearer $AUTH_BEARER_TOKEN" https://quay.io/api/v1/repository/redhat-pipeline-service/"$IMAGE_NAME")
+
+    if [[ "$digest_http_code" == "200" ]]; then
+      digest=$(jq '.tags[] | select(.name == "main").manifest_digest ' /tmp/digest.json | tr -d '"')
+      rm -f /tmp/digest.json
+      get_vulnerabilities
```

Comment:
From a code structure point of view wouldn't it make more sense that the orchestration is done from main?
meaning that the code could exit if it was not possible to get the digest and get_vulnerabilities would then get called from main.

Reply:
GitHub actions doesn't play nice with exit codes. Would lead to increase in complexity between jobs and steps.

Initially I had written the logic in main but introduced a function to keep it modular. And we don't want to call get_vulnerabilities unless we have a 200 from digest.

Reply:
exit code would have made the handling less verbose but if it is not good with github action it does not prevent you to use return and break the loop making main aware that the normal flow should be interrupted and the error flow should be executed.

Reply:
Fixed.